### ISSUE TITLE
Is it possible to disable a lint for all macro invocations?

### ISSUE THREAD

**Issue by pitdicker** [rust-lang/rust-clippy#11472]
### Description

Is there a way to suppress a lint (`zero_prefixed_literal` in my case) for all invocations of a macro?

For chrono (https://github.com/chronotope/chrono/pull/1270) I am trying to write macros that can be used as
```rust
time!(7:03:25);
```

But that generates warnings (https://github.com/chronotope/chrono/actions/runs/6119764599/job/16610345590?pr=1270)

I am looking for a solution that doesn't require users to disable the lint. Maybe it can be solved by adjusting the macro, I just don't know enough about this part of Rust yet.

**Comment by Centri3** [comment]
You can have it expand to
```rust
#[allow(clippy::zero_prefixed_literal)]
{
		(actual code)
}
```
However, it's worth noting that this lint already checks if it's in an external macro (i.e., macro outside the current crate) and doesn't lint if so. So this will only come up for chrono in particular.

**Comment by Alexendoo** [comment]
The literals aren't coming from an expansion so it'd still be linted in other crates

**Comment by Alexendoo** [comment]
I don't think so, but there is a trick to get rid of an indentation level for the allow attribute. Since the tail expression of a block can have attributes you can do

```rust
    ($y:literal-$m:literal-$d:literal) => {{
        #[allow(clippy::zero_prefixed_literal)]
        match $crate::NaiveDate::from_ymd_opt($y, $m, $d) {
            Some(d) => d,
            None => panic!("invalid calendar date"),
        }
    }};
```

**Comment by pitdicker** [comment]
This is perfect, thank you very much!
